examples/with-auth/components/session.js

```diff
@@ -0,0 +1,204 @@
+/**
+ * A class to handle signing in and out and caching session data in sessionStore
+ *
+ * Note: We usewindow.XMLHttpRequest() here rather than fetch because fetch() uses
+ * Service Workers and they cannot share cookies with the browser session
+ * yet (!) so if we tried to get or pass the CSRF token it would mismatch.
+ */
+export default class Session {
+  constructor ({req} = {}) {
+    this._session = {}
+    this._user = {}
+    try {
+      if (req) {
+        // If running on server we can access the server side environment
+        this._session = {
+          csrfToken: req.connection._httpMessage.locals._csrf
```

Comment:
it seems that `getCsrfToken()` is always called before a POST request is made, so is this line necessary? Why store the csrfToken if every time we just update the csrfToken regardless?

it might be worth using something like axios where we can use an interceptor:
```
axios.interceptors.request.use(function(config) {
  return new Promise(async function(resolve) {
    if (config.method === 'post') {
      const csrfToken = await getCsrfToken();
      config.headers = {
        ...config.headers,
        'csrf-token': csrfToken,
      };
    }
    resolve(config);
  });
});
```